cpp/src/arrow/compute/kernels/vector_selection.cc

```diff
@@ -58,2119 +60,22 @@ using internal::OptionalBitIndexer;
 namespace compute {
 namespace internal {
 
+using FilterState = OptionsWrapper<FilterOptions>;
+using TakeState = OptionsWrapper<TakeOptions>;
+
 int64_t GetFilterOutputSize(const ArraySpan& filter,
                             FilterOptions::NullSelectionBehavior null_selection) {
```

Comment:
Is this function still useful, since it's a trivial wrapper around `GetBitmapFilterOutputSize`?

Reply:
It becomes this in #35750.
```cpp
int64_t GetFilterOutputSize(const ArraySpan& filter,
                            FilterOptions::NullSelectionBehavior null_selection) {
  if (filter.type->id() == Type::BOOL) {
    return GetBitmapFilterOutputSize(filter, null_selection);
  }
  return GetREEFilterOutputSize(filter, null_selection);
```

 I will move it completely to `vector_selection_filter.cc` now and remove it here, but I haven't decided yet if the REEx* filter kernels are going to be implemented in `vector_selection_filter.cc` or in a separate `.cc`, so I might undo this later.